packages/api/src/graphql/mutation.public.ts

```diff
@@ -148,20 +149,22 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
       type: GraphQLNonNull(GraphQLPublicComment),
       args: {input: {type: GraphQLNonNull(GraphQLPublicCommentInput)}},
       description: 'This mutation allows to add a comment. The input is of type CommentInput.',
-      async resolve(_, {input}, {optionalAuthenticateUser, dbAdapter, challenge}) {
+      async resolve(_, {input}, {optionalAuthenticateUser, dbAdapter, challenge, loaders}) {
         const user = optionalAuthenticateUser()
         let authorType = CommentAuthorType.VerifiedUser
         const commentLength = countRichtextChars(0, input.text)
 
         if (commentLength > MAX_COMMENT_LENGTH) {
           throw new CommentLengthError()
         }
+        const guestCanCommentSetting = await dbAdapter.setting.getSetting(
```

Comment:
I would redo this a tiny bit for readability:

ala:
```typescript
        const canGuestsComment = (await dbAdapter.setting.getSetting(
          SettingName.ALLOW_GUEST_COMMENTING
        ))?.value
```

The if below would then read:

```typescript
if (!canGuestsComment) throw new AnonymousCommentsDisabledError()
```

Reply:
I can't access value in the first statement because it's still a promise.

Reply:
By wrapping the await in braces you can